csharp/src/Microsoft.ML.OnnxRuntime/Training/TrainingSession.shared.cs

```diff
@@ -410,6 +579,24 @@ private IntPtr[] GetOrtValuesHandles(IReadOnlyCollection<FixedBufferOnnxValue> v
             return valuesArray;
         }
 
+        private IntPtr[] ConvertNamesToUtf8(IReadOnlyCollection<string> names, DisposableList<IDisposable> cleanupList)
```

Comment:
Does cleanupList need to be passed by reference?

Reply:
Leads to compiler error:

```cs
Cannot use 'cleanupList' as a ref or out value because it is a 'using variable'
```

Seems like variables defined in the `using` block are read only. Which would imply that they are by default passed as refs? I read this here: https://stackoverflow.com/a/11636545